classes/Tools.php

```diff
@@ -4309,6 +4311,46 @@ public static function isFileFresh(string $file, int $timeout = self::CACHE_LIFE
 
         return false;
     }
+
+    /**
+     * @return bool
+     */
+    public static function isCountryFromBrowserAvailable(): bool
+    {
+        $languageAvailable = self::getCountryIsoCodeFromHeader();
+
+        if ($languageAvailable === null) {
+            return false;
+        }
+
+        return Configuration::get('PS_DETECT_COUNTRY')
+            && self::isCountryIsoCodeAvailableInHeader()
+            && Validate::isLanguageIsoCode($languageAvailable);
+    }
+
+    /**
+     * @return bool
+     */
+    public static function isCountryIsoCodeAvailableInHeader(): bool
+    {
+        $isLanguageInHeader = static::getCountryIsoCodeFromHeader();
+
+        return $isLanguageInHeader !== null;
+    }
+
+    /**
+     * @return string|null
+     */
+    public static function getCountryIsoCodeFromHeader(): ?string
+    {
+        if (isset($_SERVER['HTTP_ACCEPT_LANGUAGE'])) {
+            return null;
+        }
+
+        preg_match(self::LANGUAGE_EXTRACTOR_REGEXP, $_SERVER['HTTP_ACCEPT_LANGUAGE'], $languages);
```

Comment:
```suggestion
        preg_match(static::LANGUAGE_EXTRACTOR_REGEXP, $_SERVER['HTTP_ACCEPT_LANGUAGE'], $languages);
```

Reply:
You're totaly right, it may be dangerous in case of extend.